jcommune-service/src/main/java/org/jtalks/jcommune/service/nontransactional/LocationService.java

```diff
@@ -82,6 +82,19 @@ public LocationService(UserService userService, SessionRegistry sessionRegistry)
      * topic/branch viewer's list until explicitly added
      */
     public void clearUserLocation() {
-        registerUserMap.remove(userService.getCurrentUser());
+        UserInfo currentUser = securityService.getCurrentUserInfo();
+        clearUserLocation(currentUser);
+    }
+
+    /**
+     * Clears forum location for the given user. After the call, user
+     * passed as param will be excluded from all the topic/branch viewer's list
+     *
+     * @param user to clean location.
+     */
+    public void clearUserLocation(Object user) {
+        if (user instanceof UserInfo) {
```

Comment:
if key is null - NullPointerException will be thrown, or am i wrong? I could check for null but I think it's more pretty.